### ISSUE TITLE
Error in build: can't resolve reference... with Node 10.x

### ISSUE THREAD

**Issue by wohali** [apache/couchdb-fauxton#1229]
Trying to build Fauxton with node 10.16.3. Getting unsolveable errors.

Error follows:

```
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! fauxton@1.2.1 webpack:release: `webpack --optimize-minimize --debug --progress --colors --config ./webpack.config.release.js`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the fauxton@1.2.1 webpack:release script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/jenkins/.npm/_logs/2019-10-10T20_54_24_219Z-debug.log
 Use --force to continue.
```

On `ppc64le` we get a slightly different error trace:
```npm
...
> uglifyjs-webpack-plugin@0.4.6 postinstall /home/jenkins/couchdb/src/fauxton/node_modules/uglifyjs-webpack-plugin
> node lib/post_install.js

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"ppc64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules/babel-cli/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"ppc64"})

added 1938 packages from 946 contributors and audited 30565 packages in 376.836s
found 1228 vulnerabilities (92 low, 65 moderate, 1068 high, 3 critical)
  run `npm audit fix` to fix them, or `npm audit` for details
Running "clean:release" (clean) task
>> 0 paths cleaned.

Running "get_deps:default" (get_deps) task
Fetching external dependencies
0 local dependencies

Running "gen_load_addons:default" (gen_load_addons) task

Running "gen_initialize:release" (gen_initialize) task

Running "copy:distDepsRequire" (copy) task
Created 12 directories, copied 218 files

Running "shell:webpackrelease" (shell) task

> fauxton@1.1.20 webpack:release /home/jenkins/couchdb/src/fauxton
> webpack --optimize-minimize --debug --progress --colors --config ./webpack.config.release.js


/home/jenkins/couchdb/src/fauxton/node_modules/webpack/node_modules/ajv/lib/ajv.js:356
    throw e;
    ^
Error: can't resolve reference #/definitions/nonNegativeInteger from id http://json-schema.org:5.104235503814077e+38/draft-07/schema#
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! fauxton@1.1.20 webpack:release: `webpack --optimize-minimize --debug --progress --colors --config ./webpack.config.release.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the fauxton@1.1.20 webpack:release script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/jenkins/.npm/_logs/2019-10-09T00_00_58_353Z-debug.log

Warning: Command failed: npm run webpack:release
```

Logfiles attached:
* [2019-10-10T20_54_24_219Z-debug.log](https://github.com/apache/couchdb-fauxton/files/3714592/2019-10-10T20_54_24_219Z-debug.log)
* [2019-10-09T00_00_58_353Z-debug.log](https://github.com/apache/couchdb-fauxton/files/3704990/2019-10-09T00_00_58_353Z-debug.log) (`ppc64le`)

Is this related to #1230 and #1233 ?

**Comment by willholley** [comment]
This looks suspiciously similar to https://github.com/apache/couchdb-fauxton/issues/1230 - worth testing again when https://github.com/apache/couchdb/pull/2250 is merged.

**Comment by willholley** [comment]
bah - I'm still observing problems on CentOS. It seems like something to do with peer dependencies; no idea why it only occurs on CentOS and not other platforms.

Running locally I see the following warnings when attempting to update / audit packages:
```
npm WARN acorn-jsx@5.0.2 requires a peer of acorn@^6.0.0 || ^7.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN ajv-keywords@3.4.1 requires a peer of ajv@^6.9.1 but none is installed. You must install peer dependencies yourself.
```

However, these don't prevent the release build from working.

@popojargo @Antonio-Maranhao any thoughts?

**Comment by wohali** [comment]
@willholley see above, and I saw your removed comments in email.  I manually applied the fixes in #1233 and builds are once again succeeding on Node 10.x in debian and centos on x86_64.

I'm rebuilding the ppc64le image now for unrelated reasons (it's very, very slow to build) and will check fauxton building on it with the same patch.

**Comment by wohali** [comment]
Splitting the `ppc64le` issue out. This particular issue will be fixed by #1233.

**Comment by willholley** [comment]
Fixed in #1233. FWIW this also failed in node 8. I believe [updating webpack](https://github.com/apache/couchdb-fauxton/commit/414769a680c113312059d407008f7fe9bc961fec) is the breaking change and just hadn't been exercised in CI until we cut a Fauxton release.
